StaticAnalyzer/views/android/VirusTotal.py

```diff
@@ -21,7 +21,16 @@ def get_report(self, file_hash):
             }
             headers = {"Accept-Encoding": "gzip, deflate"}
             try:
-                response = requests.get(url, params=params, headers=headers)
+                if settings.UPSTREAM_PROXY_ENABLED:
+                    if not settings.UPSTREAM_PROXY_USERNAME:
+                        proxy_port = str(settings.UPSTREAM_PROXY_PORT)
+                        proxy_host = settings.UPSTREAM_PROXY_TYPE + '://'  + settings.UPSTREAM_PROXY_IP + ':' + proxy_port
+                        proxies = {"https": proxy_host}
+                    else:
+                        proxy_port = str(settings.UPSTREAM_PROXY_PORT)
+                        proxy_host = settings.UPSTREAM_PROXY_TYPE + '://' + settings.UPSTREAM_PROXY_USERNAME + ':' + settings.UPSTREAM_PROXY_PASSWORD + "@" + settings.UPSTREAM_PROXY_IP + ':' + proxy_port
+                        proxies = {"https": proxy_host}

```

Comment:
Should this "https" be `settings.UPSTREAM_PROXY_TYPE` ?

Reply:
in proxy type you set-up if you proxy is over https or http .
{"https": proxy_host}   mean you forward https traffic to the proxy